spring-cloud-aws-autoconfigure/src/main/java/org/springframework/cloud/aws/autoconfigure/context/ContextRegionProviderAutoConfiguration.java

```diff
@@ -30,6 +31,7 @@
  * @author Agim Emruli
  */
 @Configuration
+@ConditionalOnAwsCloudEnvironment
```

Comment:
This one is wrong as well.
See its source code:
```
registerRegionProvider(registry, this.environment.getProperty("cloud.aws.region.auto", Boolean.class, true) &&
	!(this.environment.containsProperty("cloud.aws.region.static")),
	this.environment.getProperty("cloud.aws.region.static"));
```
So, we register `RegionProvider` if it is `auto` - `Ec2MetadataRegionProvider`, or if we explicitly provide `cloud.aws.region.static`.
Otherwise there will no any `RegionProvider` independently of the environment.